x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -62,7 +63,11 @@ impl Program {
 
         let args_bytes = borsh::to_vec(&args).map_err(|_| StateError::Serialization)?;
 
-        Ok(unsafe { ffi(args_bytes.as_ptr(), args_bytes.len()) })
+        let ptr = unsafe { ffi(args_bytes.as_ptr(), args_bytes.len()) };
+
+        let bytes = into_bytes(ptr).unwrap_or_default();
```

Comment:
Should we not return an `Err(Error::InvalidPointer)` here instead of deserializing from `None` ?

Reply:
That's a great question. Again, I can't imagine a scenario where we return an invalid pointer where there's a recoverable error. If that does happen, the code will panic because deserialization will fail. 

I think we'll have to spend some time testing these edge cases, but we should do that with the new host-code that hasn't been merged yet.